### PR TITLE
[HOLD] Add "engines" field, .nvmrc and .npmrc to enforce a specific node version

### PR DESCRIPTION
**Pull request by HorusGoul**
cc @cead22 @stitesExpensify 

### Details
This would help us enforce the node version for both internal and external contributors. 

### Fixed Issues

$ GH_LINK

### Tests/QA Tests
1. Open a terminal in the project's dir
2. Run `nvm install 12`. This will set your node version to 12.
3. Run `npm install`. It should fail, telling you that you're not using the node version specified in the `engines.node` field of the `package.json` file.
4. Run `nvm use`. This will set your node version to the one defined in the `.nvmrc` file. It should be the same version as the one in the `engines.node` field.
5. Run `npm install`. It should install all the packages.

### CODE REVIEW

**Review by Dal-Papa — DISMISSED** [review]

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+14.17.4
```
Comment (TomatoToaster) [inline .nvmrc:1]: NAB: Could we add a NewLine to end of these files?

**Review by TomatoToaster — DISMISSED** [review]
This is neat!

**Comment by TomatoToaster** [comment]
NAB: Since we have this now, do we want to also add some sort of guidelines to use `nvm` in our Readme?

**Review by cead22 — DISMISSED** [review]
Thanks for this, I updated the SOs to include App
- https://stackoverflow.com/c/expensify/posts/6259/revisions
- https://stackoverflow.com/c/expensify/posts/5957/revisions

File: .nvmrc (hunk 1 above)
Comment (HorusGoul) [reply .nvmrc]: Added!

**Comment by HorusGoul** [comment]
Should we keep this on hold until after N6?

**Review by Dal-Papa — APPROVED** [review]

**Review by TomatoToaster — APPROVED** [review]

**Review by stitesExpensify — APPROVED** [review]

**Review by tgolen — APPROVED** [review]

**Review by cead22 — APPROVED** [review]

**Comment by tgolen** [comment]
Can the hold be removed now and this merged? Looks like there is a conflict.

**Comment by cead22** [comment]
I think so, but 👃 on taking this over and resolve the conflicts, since my plate is kinda full

**Comment by tgolen** [comment]
I'm just gonna close this for now since it looks like no one wants to take it over.

**Pull request closed without merging**